lib/ansible/module_utils/rds.py

```diff
@@ -0,0 +1,135 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible. If not, see <http://www.gnu.org/licenses/>.
+
+import botocore
+import time
+
+from ansible.module_utils.ec2 import boto3_tag_list_to_ansible_dict
+
+
+def get_db_instance(conn, instancename):
+    try:
+        response = conn.describe_db_instances(DBInstanceIdentifier=instancename)
+        instance = RDSDBInstance(response['DBInstances'][0])
+        return instance
+    except botocore.exceptions.ClientError as e:
+        if e.response['Error']['Code'] == 'DBInstanceNotFound':
+            return None
+        else:
+            raise
+
+
+def get_db_snapshot(conn, snapshotid):
+    try:
+        response = conn.describe_db_snapshots(DBSnapshotIdentifier=snapshotid)
+        snapshot = RDSSnapshot(response['DBSnapshots'][0])
+        return snapshot
+    except botocore.exceptions.ClientError:
+        if e.response['Error']['Code'] == 'DBSnapshotNotFound':
+            return None
+        else:
+            raise
+
+
+class RDSDBInstance(object):
+    def __init__(self, dbinstance):
+        self.instance = dbinstance
+        if 'DBInstanceIdentifier' not in dbinstance:
+            self.name = None
+        else:
+            self.name = self.instance.get('DBInstanceIdentifier')
+        self.status = self.instance.get('DBInstanceStatus')
+        self.data = self.get_data()
+
+    def get_data(self):
+        d = {
+            'id': self.name,
+            'create_time': self.instance.get('InstanceCreateTime', ''),
+            'db_engine': self.instance['Engine'],
+            'db_name': self.instance.get('DBName'),
+            'status': self.status,
+            'availability_zone': self.instance.get('AvailabilityZone'),
+            'backup_retention': self.instance['BackupRetentionPeriod'],
+            'maintenance_window': self.instance['PreferredMaintenanceWindow'],
+            'multi_zone': self.instance['MultiAZ'],
+            'instance_type': self.instance['DBInstanceClass'],
+            'username': self.instance['MasterUsername'],
+            'replication_source': self.instance.get('ReadReplicaSourceDBInstanceIdentifier'),
+            'size': self.instance.get('AllocatedStorage'),
+            'storage_type': self.instance.get('StorageType')
+        }
+        if self.instance.get('Iops'):
+            d['iops'] = self.instance['Iops'],
+        if self.instance["VpcSecurityGroups"] is not None:
+            d['vpc_security_groups'] = ','.join(x['VpcSecurityGroupId'] for x in self.instance['VpcSecurityGroups'])
+        if self.instance.get("Endpoint"):
+            d['endpoint'] = self.instance["Endpoint"].get('Address', None)
+            d['port'] = self.instance["Endpoint"].get('Port', None)
```

Comment:
Missing HostedZoneId here which is important if you want to set a route53 ALIAS record later